launch/dynamo-run/src/subprocess/trtllm_inc.py

```diff
@@ -253,16 +487,61 @@ def cmd_line_args():
     parser.add_argument(
         "--publish-events-and-metrics",
         action="store_true",
-        help="Publish events and metrics to the dynamo components.",
+        help="Publish events and metrics to the dynamo components. Note: This is not supported when running in prefill disaggregation mode.",
+    )
+    parser.add_argument(
+        "--task",
+        type=str,
+        action="append",
+        choices=["prefill", "decode", "prefill_and_decode"],
+        default=[],
+        help="Specifies the task for the engine. Can be specified multiple time for different tasks. Will raise an error if conflicting tasks are specified.",
+    )
+    parser.add_argument(
+        "--remote-prefill-endpoint",
+        type=str,
+        default=DEFAULT_PREFILL_ENDPOINT,
+        help=f"Endpoint(in 'dyn://namespace.component.endpoint' format) to send prefill requests to when running in decode disaggregation mode. Default: {DEFAULT_PREFILL_ENDPOINT}",
     )
     args = parser.parse_args()
 
+    # Validate arguments
     if args.context_length is not None:
         warnings.warn(
             "--context-length is accepted for compatibility but will be ignored for TensorRT-LLM. Please provide max_input_len, max_seq_len and max_output_len in yaml file and point --extra-engine-args to the yaml file.",
             UserWarning,
         )
 
+    endpoint = args.endpoint
+
+    # disaggregation mode
+    disaggregation_mode = None
+    for choice in ["prefill", "decode", "prefill_and_decode"]:
+        if choice in args.task:
```

Comment:
FYI - argparse `choices` will prevent you from passing anything other than the choices, so you shouldn't be able to pass in anything other than this list. 

You could also probably set `default="prefill_and_decode"` for now and remove need for the `None` handling logic.

Reply:
If setting 'default="prefill_and_decode"', the --task option would add another field to the list causing conflicting specification.
We can keep adding the choices for different attributes that the engine can perform.